src/compiler/scala/tools/nsc/backend/jvm/GenBCode.scala

```diff
@@ -329,17 +329,17 @@ abstract class GenBCode extends BCodeSyncAndTry {
      *
      */
     override def run() {
-      val bcodeStart = statistics.startTimer(bcodeTimer)
+      val bcodeStart = if (statistics.canEnable) statistics.startTimer(bcodeTimer) else null
```

Comment:
It seems to me that it would make more sense to do the opposite: make sure that all operations in `Statistics` are guarded by `canEnable` checks, and remove all the existing `canEnable` checks at the callsites. It's too easy to forget adding the check when introducing a new metric.